src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2SubmissionTest.kt

```diff
@@ -646,6 +651,15 @@ class Cas2SubmissionTest : IntegrationTestBase() {
               .expectStatus()
               .isOk
           }
+
+          // verify that generated 'application.submitted' domain event links to the CAS2 domain
+          val persistedDomainEvent = domainEventRepository.findFirstByOrderByCreatedAtDesc()
+          val domainEventFromJson = objectMapper.readValue(
+            persistedDomainEvent!!.data,
+            Cas2ApplicationSubmittedEvent::class.java,
+          )
+          Assertions.assertThat(domainEventFromJson.eventDetails.applicationUrl)
+            .matches("http://cas2.frontend/applications/.+")
```

Comment:
This makes me a tad nervous because it's linked to the configuration. It might be better to set a variable in the class to get the value of `url-templates.frontend.cas2.application` and check the value is correct.  Something like:

```
@Value("\${url-templates.frontend.cas2.application}") 
private val applicationUrlTemplate: String
```

In the body of the test class, and then:

```kotlin
val expectedUrl = applicationUrlTemplate.replace('#applicationId', application.id)
...
Assertions.assertThat(domainEventFromJson.eventDetails.applicationUrl).isEqualTo(expectedUrl)
```

I have no idea if that would work, but just a thought